frontend/src/metabase/lib/dom.js

```diff
@@ -272,21 +280,20 @@ export function open(
     // custom function for opening in new window
     openInBlankWindow = url => clickLink(url, true),
     // custom function for opening in same app instance
-    openInSameOrigin = openInSameWindow,
+    openInSameOrigin,
     ignoreSiteUrl = false,
     ...options
   } = {},
 ) {
-  const isOriginalUrlAbsolute = isAbsoluteUrl(url); // this does not check real "absolute" url, but if a url should be resolved from the root URL
   url = ignoreSiteUrl ? url : getWithSiteUrl(url);
 
   if (shouldOpenInBlankWindow(url, options)) {
     openInBlankWindow(url);
   } else if (isSameOrigin(url)) {
-    if (isOriginalUrlAbsolute) {
+    if (!isMetabaseUrl(url)) {
```

Comment:
The original condition `isOriginalUrlAbsolute` causes the problem where we want to navigate to questions or dashboards via click behavior because the `isOriginalUrlAbsolute` at this point would be `true` from the url `/question/xxx` or `/dashboard/xxx`, and we'll call `clickLink` which doesn't use client-side routing, so all state in redux store is lost. And that causes a problem when we're embedding with UI parameters e.g. `top_nav=false` because the UI would be reset back to the default value instead.
https://github.com/metabase/metabase/blob/43639a9cfcc4fc16c55bf8545e5f2d34a4c90551/frontend/src/metabase/redux/embed.ts#L7-L17